pkg/image/docker/save.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright the Sonobuoy contributors 2019
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	log "github.com/sirupsen/logrus"
+
+	"github.com/heptio/sonobuoy/pkg/image/exec"
+)
+
+// Save exports a set of images to a tar file
+func Save(images []string, filename string) error {
+	log.Info("Saving images: ...")
+
+	// Build out docker command
+	args := append([]string{"save"}, images...)
+	args = append(args, "--output", filename)
+
+	err := exec.Command("docker", args...).Run()
+	// retry pushing up to retries times if necessary
+	if err != nil {
+		log.WithError(err).Infof("Failed to save images.")
```

Comment:
nit: I think the pattern should probably be that we just return the wrapped error rather than logging it _and_ returning it?

That way the caller can decide how to handle it (log or not) and we don't end up with repetitive logs where each level logs and returns.

Reply:
Example here (reminder this is just a nit and maybe we actually prefer this?):
```
INFO[0043] Pushing image: gcr.io/google-samples/gb-frontend:v6 ...
INFO[0047] Trying again to push image: gcr.io/google-samples/gb-frontend:v6 ...  error="exit status 1"
INFO[0050] Failed to push image: gcr.io/google-samples/gb-frontend:v6  error="exit status 1"
ERRO[0050] couldn't push image: gcr.io/google-samples/gb-frontend:v6: exit status 1
```

It is a little verbose but it does I suppose make it very clear what was happening. Again, the error message itself is useless since its just the code.

Reply:
I think the retry is good to show that it's trying again. Let me see if I can get the real error returned like you suggested in another comment.